### PR TITLE
Add `FourierTransform`

### LINKED ISSUE #292: `FourierTransform`
### 🚀 Feature Request

Add `FourierTransform`, that creates fourier features.

### Motivation

It can help to deal with multiple and non-integer seasonalities.

### Proposal

Add `FourierTransform`, that creates fourier features.

Parameters:
* period: float: the period of the seasonality to capture in frequency units of time series (must be >= 2, can be float to support not regular seasonality, for example year seasonality in weekly data); 
* order: int: number of Fourier components to include (must be <= 2*period).

Fourier k-th component:
```latex
s_k(t) = \sin \left( 2 pi k \frac{t}{p f} ​\right)
c_k(t) = \cos \left( 2 pi k \frac{t}{p f} ​\right)
```
where `p` - period, `f` - freq of time series, `t` - time from the "zero" of the period (can be the beginning of the series).

It will be better to do it without `PerSegmentWrapper`.

Use [Fourier](https://www.statsmodels.org/devel/generated/statsmodels.tsa.deterministic.Fourier.html) and [CalendarFourier](https://www.statsmodels.org/devel/generated/statsmodels.tsa.deterministic.CalendarFourier.html#statsmodels.tsa.deterministic.CalendarFourier) for reference.

To take care of names of the columns we want to create more precise object constructor, that can create one particular column (its repr will be the default name of the column). For example, it will take in:
* period
* mods: list of components, that distinguishes sin and cos.

For example `mods=[0, 1, 2]` creates `s_1, c_1, s_2`.

The constructor with `order` can reuse this precise constructor.

Parameter `out_column` is a prefix name for all the columns and single columns are separated by _{number} suffix (like in `LagTransform`).

### Test cases

1. Columns with expected names are added with `out_column` and without `out_column` parameter. (It can be better to test general and specific constructors separately).
2. Values are expected (test integer and float periods).
3. Constraints are satisfied for `period` and `mods` (may be be possible to also check `order` here, but it can be managed inside constructor with `mods`).
4. Sanity check example with two seasonalities. We are going to catch them by using only this transform two times.

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [x] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by Mr-Geekman**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes
See #292. But there was some change in constructors and mods indexation.

## Related Issue
#292.

## Closing issues
Closes #292.

### CODE REVIEW

File: etna/transforms/fourier.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        ----------
+        period:
+            the period of the seasonality to capture in frequency units of time series, it should be >= 2
+        order:
+            upper order of Fourier components to include, it should be >= 1 and <= ceil(period/2))
+        mods:
+            alternative and precise way of defining which harmonics will be used,
+            for example `mods=[1, 3, 4]` means that sin of the first order
+            and sin and cos of the second order will be used,
+            mods should be >= 1 and < period
```
Comment (iKintosh) [inline etna/transforms/fourier.py:33]: Sorry, but it is really difficult to understand without example =(
What if I have daily data and want to capture seasonality of 15 days. What should my parameters be?

**Review by iKintosh — CHANGES_REQUESTED** [review]
Other than the comment it looks great!

File: etna/transforms/fourier.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/transforms/fourier.py:33]: I added Notes section to describe the behavior.

**Review by iKintosh — APPROVED** [review]
Thanks 👍

**Pull request merged**